.docker/tier1/Dockerfile

```diff
@@ -6,6 +6,8 @@ RUN mkdir /judge /problems && cd /judge && \
 	pip3 install -e . && \
 	HOME=~judge . ~judge/.profile && \
 	runuser -u judge -w PATH -- dmoj-autoconf -V > /judge-runtime-paths.yml && \
-	echo '  crt_x86_in_lib32: true' >> /judge-runtime-paths.yml
+	echo '  crt_x86_in_lib32: true' >> /judge-runtime-paths.yml && \
+	echo 'problem_storage_root: [ "/problems" ]' > /judge.yml && \
```

Comment:
Why? We already use `/problems` for problem storage by default without a configuration file. Your `/judge.yml` literally reimplements the existing default behaviour.

Reply:
I am not sure, but when I try to run this command

```shell
docker run \
  --name judge \
  -v /mnt/problems:/problems \
  --cap-add=SYS_PTRACE \
  -d \
  dmoj/judge-tier3:latest \
  run $BRIDGE_ADDRESS $JUDGE_NAME $JUDGE_KEY
```
I get an error. You can see the logs [here](https://github.com/osmanirosado/judge-server/tree/logs/logs).

Note that the `-c` option appears as mandatory in the documentation show by the read me file of the project.

Reply:
`-c` is not necessary in docker. Also this looks suspiciously like a shell quoting issue that you managed to work around by using yaml. The real solution is to properly quote your key.

Reply:
The environment variables are quoted now and the command `docker-compose config` is used to review if the variable interpolation is correct, but the error persists. I share the code [here](https://github.com/osmanirosado/dmoj-judge-server/blob/develop/docker-compose.yml).  You can see the logs bellow.

```
app_1  | Running live judge...
app_1  | INFO 2021-08-10 19:09:58,139 55 packet Preparing to connect to [10.12.101.21]:9999 as: None
app_1  | INFO 2021-08-10 19:09:58,139 55 packet TLS not enabled.
app_1  | INFO 2021-08-10 19:09:58,160 55 packet Opening connection to: [10.12.101.21]:9999
app_1  | INFO 2021-08-10 19:09:58,163 55 packet Starting handshake with: [10.12.101.21]:9999
app_1  | INFO 2021-08-10 19:09:58,165 55 packet Awaiting handshake response: [10.12.101.21]:9999
app_1  | ERROR 2021-08-10 19:09:58,172 55 packet Cannot understand handshake response: [10.12.101.21]:9999
app_1  | Traceback (most recent call last):
app_1  |   File "/judge/dmoj/packet.py", line 288, in handshake
app_1  |     size = PacketManager.SIZE_PACK.unpack(data)[0]
app_1  | struct.error: unpack requires a buffer of 4 bytes
app_1  | ERROR 2021-08-10 19:09:58,173 55 packet Authentication as "None" failed on: [10.12.101.21]:9999
app_1  | WARNING 2021-08-10 19:09:58,173 55 packet Attempting reconnection in 4s: [10.12.101.21]:9999
app_1  | INFO 2021-08-10 19:09:58,173 55 packet Dropping old connection.
```

Reply:
Is your judge called `None`? If not, there has to be a variable interpolation issue somewhere, like `$JUDGE_NAME` somehow being an empty string.

Reply:
I have inserted `set -x`, before the last command `runuser -u judge "${command[@]}" -- "$@"` of the [entry](https://github.com/osmanirosado/dmoj-judge-server/blob/e9a67272fb08657ea323115c59a0a4eab7423332/entry#L24) point script, to see the executed code. I build the image again and set up the judge. This is the result:

```
Creating network "judge1_default" with the default driver
Creating judge1_app_1 ... done
Attaching to judge1_app_1
app_1  | /home/judge/.profile: line 31: /root/.cargo/env: No such file or directory
app_1  | + runuser -u judge dmoj -- 10.12.101.21 faraday-judge-1 'qE-T^)Ui3~*{AI(j-Qi:n*kC)pd#7dQ8N;Ae6u+U>%s;l)&}?w0@%44^AXuob[:c%lL-#]RB:68e+V!z)X}!sgN@,Mep4ww8zxhN'
app_1  | Self-testing executors
app_1  | Self-testing ADA:    Success [0.003s, 1736 KB]   gnatmake 10.2.1
```

In the output, we can see the command. The environment variables are correctly interpolated. But I get the same error:

```
app_1  | Running live judge...
app_1  | INFO 2021-08-11 14:13:14,116 55 packet Preparing to connect to [10.12.101.21]:9999 as: None
app_1  | INFO 2021-08-11 14:13:14,116 55 packet TLS not enabled.
app_1  | INFO 2021-08-11 14:13:14,136 55 packet Opening connection to: [10.12.101.21]:9999
app_1  | INFO 2021-08-11 14:13:14,139 55 packet Starting handshake with: [10.12.101.21]:9999
app_1  | INFO 2021-08-11 14:13:14,141 55 packet Awaiting handshake response: [10.12.101.21]:9999
app_1  | ERROR 2021-08-11 14:13:14,149 55 packet Cannot understand handshake response: [10.12.101.21]:9999
```

I also note an error before the command execution, the line 31 of `.profile` script fails.